server/monitor-types/rabbitmq.js

```diff
@@ -0,0 +1,74 @@
+const { MonitorType } = require("./monitor-type");
+const { log, UP, DOWN } = require("../../src/util");
+const { axiosAbortSignal } = require("../util-server");
+const axios = require("axios");
+
+class RabbitMqMonitorType extends MonitorType {
+    name = "rabbitmq";
+
+    /**
+     * @inheritdoc
+     */
+    async check(monitor, heartbeat, server) {
+        // HTTP basic auth
+        let basicAuthHeader = {};
+        basicAuthHeader = {
+            "Authorization": "Basic " + this.encodeBase64(monitor.rabbitmqUsername, monitor.rabbitmqPassword),
+        };
+
+        let status = DOWN;
+        let msg = "";
+
+        for (const baseUrl of JSON.parse(monitor.rabbitmqNodes)) {
+            try {
+                const options = {
+                    url: new URL("/api/health/checks/alarms", baseUrl).href,
+                    method: "get",
+                    timeout: monitor.timeout * 1000,
+                    headers: {
+                        "Accept": "application/json",
+                        ...(basicAuthHeader),
+                    },
+                    signal: axiosAbortSignal((monitor.timeout + 10) * 1000),
+                    validateStatus: () => true,
```

Comment:
The signal is for connection timeouts. Added a comment for this.
validateStatus was originally intended to prevent exception on 404 and handle it with if clause. Versions before 3.8.10 (released on Jan 21, 2021) do not have the alarms endpoint used in this PR. I thought it became too complex and removed the if part though.